tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/reindexing/ElasticsearchCMISPathTests.java

```diff
@@ -0,0 +1,127 @@
+    @Autowired
+    private DataContent dataContent;
+    @Autowired
+    private DataSite dataSite;
+    @Autowired
+    private ServerHealth serverHealth;
+    @Autowired
+    protected SearchQueryService searchQueryService;
+
+    private UserModel user;
+    private SiteModel siteModel;
+    private FolderModel folder0;
+    private FolderModel folder00;
+    private FolderModel folder000;
+    private FolderModel folder1;
+    private FileModel document0000;
+    private FileModel document00;
+    private FileModel document01;
+    private FileModel document10;
+
+    /**
+     * Data will be prepared using the schema below:
+     * <pre>
+     * Site
+     * + Document Library
+     *   +-folder0
+     *   | +-folder00
+     *   | | +-folder000
+     *   | |   +-document0000
+     *   | +-document00
+     *   | +-document01
+     *   +-folder1
+     *   +-document10
+     * </pre>
+     */
+    @BeforeClass (alwaysRun = true)
+    public void dataPreparation()
+    {
+        serverHealth.assertServerIsOnline();
+
+        Step.STEP("Index system nodes.");
+        AlfrescoStackInitializer.reindexEverything();
+
+        user = dataUser.createRandomTestUser();
+
+        siteModel = dataSite.usingUser(user).createPrivateRandomSite();
+
+        folder0 = dataContent.usingUser(user).usingSite(siteModel).createFolderCmisApi(FOLDER_0_NAME);
+        folder1 = dataContent.usingUser(user).usingSite(siteModel).createFolderCmisApi(FOLDER_1_NAME);
+
+        folder00 = dataContent.usingUser(user).usingResource(folder0).createFolderCmisApi(FOLDER_00_NAME);
+        folder000 = dataContent.usingUser(user).usingResource(folder00).createFolderCmisApi(FOLDER_000_NAME);
+
+        document0000 = createContent(DOC_0000_NAME, "This is document 0000", folder000, user);
+        document00 = createContent(DOC_00_NAME, "This is document 00", folder0, user);
+        document01 = createContent(DOC_01_NAME, "This is document 01", folder0, user);
+        document10 = createContent(DOC_10_NAME, "This is document 10", folder1, user);
+    }
+
+    @TestRail (description = "Check that we can select subfolders using IN_TREE with Elasticsearch.", section = TestGroup.SEARCH, executionType = ExecutionType.REGRESSION)
+    @Test (groups = TestGroup.SEARCH)
+    public void inTreeQuery_selectSubfolders()
+    {
+        SearchRequest query = req("cmis", "SELECT * FROM cmis:folder WHERE IN_TREE('" + folder0.getNodeRef() + "') AND CONTAINS('*')");
+        searchQueryService.expectResultsFromQuery(query, user, FOLDER_00_NAME, FOLDER_000_NAME);
+    }
+
+    @TestRail (description = "Check that we can select documents using IN_TREE with Elasticsearch.", section = TestGroup.SEARCH, executionType = ExecutionType.REGRESSION)
+    @Test (groups = TestGroup.SEARCH)
+    public void inTreeQuery_selectDocuments()
+    {
+        SearchRequest query = req("cmis", "SELECT * FROM cmis:document WHERE IN_TREE('" + folder0.getNodeRef() + "') AND CONTAINS('*')");
```

Comment:
As per previous comment, is the CONTAINS('*') still needed given we have CMIS query consistency set to EVENTUAL here:

https://github.com/Alfresco/acs-packaging/blob/master/tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/AlfrescoStackInitializer.java#L289